vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/overview/components/ProjectCollection.java

```diff
@@ -0,0 +1,191 @@
+    addClassNames("projects-collection");
+    layoutSearchField();
+    layoutProjectCreationButton();
+    layoutGrid();
+    layoutTitleSection();
+    layoutControlSection();
+    layoutGridSection();
+    controlSection.add(new Div(projectSearchField));
+    controlSection.add(new Div(createProjectButton));
+
+    gridSection.add(projectGrid);
+    add(titleSection);
+    add(controlSection);
+    add(gridSection);
+  }
+
+  private void createLazyProjectView() {
+    projectPreviewGridLazyDataView = projectGrid.setItems(query -> {
+      List<SortOrder> sortOrders = query.getSortOrders().stream().map(
+              it -> new SortOrder(it.getSorted(), it.getDirection().equals(SortDirection.DESCENDING)))
+          .collect(Collectors.toList());
+      // if no order is provided by the grid order by last modified (the least priority)
+      sortOrders.add(SortOrder.of("lastModified").descending());
+      return projectInformationService.queryPreview(projectPreviewFilter, query.getOffset(),
+          query.getLimit(), List.copyOf(sortOrders)).stream();
+    });
+  }
+
+  private void configureSearch() {
+    projectSearchField.setValueChangeMode(ValueChangeMode.LAZY);
+    projectSearchField.addValueChangeListener(event -> {
+      projectPreviewFilter = event.getValue().trim();
+      projectPreviewGridLazyDataView.refreshAll();
+    });
+  }
+
+  private void configureProjectCreationButton() {
+    createProjectButton.addClickListener(listener -> {
+      fireClickEvent();
+    });
+  }
+
+  private void layoutSearchField() {
+    projectSearchField.setPlaceholder("Search");
+    projectSearchField.setClearButtonVisible(true);
+    projectSearchField.setSuffixComponent(VaadinIcon.SEARCH.create());
+    projectSearchField.addClassNames("search-field");
+  }
+
+  private void layoutProjectCreationButton() {
+    createProjectButton.addClassName("add-project-button");
+  }
+
+  private void layoutGrid() {
+    projectGrid.addColumn(new ComponentRenderer<>(
+        item -> new Anchor(String.format(Projects.PROJECT_INFO, item.projectId().value()),
+            item.projectCode()))).setHeader("Code").setWidth("7em").setFlexGrow(0);
+
+    projectGrid.addColumn(new ComponentRenderer<>(
+        item -> new Anchor(String.format(Projects.PROJECT_INFO, item.projectId().value()),
+            item.projectTitle()))).setHeader("Title").setKey("projectTitle");
+
+    projectGrid.addColumn(new LocalDateTimeRenderer<>(
+        projectPreview -> asClientLocalDateTime(projectPreview.lastModified()),
+        "yyyy-MM-dd HH:mm:ss")).setKey("lastModified").setHeader("Last Modified");
+
+    projectGrid.setMultiSort(true);
+  }
+
+  private void layoutTitleSection() {
+    titleSection.addClassName("title");
+    titleSection.setText(title);
+  }
+
+  private void layoutControlSection() {
+    controlSection.addClassName("controls");
+  }
+
+  private void layoutGridSection() {
+    gridSection.addClassName("projects-grid");
+  }
```

Comment:
Hi, I would argue that those methods do not add a lot of information. 
NIT: you could inline them back.

Reply:
true, i guess that is just my style to prepare it :D